lib/ruby_lsp/ruby_lsp_rails/addon.rb

```diff
@@ -28,20 +28,31 @@ def initialize
         # the real client is initialized, features that depend on it will not be blocked by using the NullClient
         @rails_runner_client = T.let(NullClient.new, RunnerClient)
         @global_state = T.let(nil, T.nilable(GlobalState))
+        @addon_mutex = T.let(Mutex.new, Mutex)
+        @client_mutex = T.let(Mutex.new, Mutex)
+        @client_mutex.lock
+
+        Thread.new do
+          @addon_mutex.lock
+          @client_mutex.synchronize { @rails_runner_client = RunnerClient.create_client }
+          @addon_mutex.unlock
+        end
       end
 
       sig { returns(RunnerClient) }
-      attr_reader :rails_runner_client
+      def rails_runner_client
+        @addon_mutex.synchronize { @rails_runner_client }
```

Comment:
What happens if requests are sent to NullClient?

Reply:
I wonder if it's worth holding on to them and forwarding to real client but it's not a priority at all.